extensions/AzureOpenAI/AzureOpenAIConfig.cs

```diff
@@ -74,6 +74,15 @@ public enum APITypes
     /// </summary>
     public int? EmbeddingDimensions { get; set; }
 
+    /// <summary>
+    /// Old AI model like ada (https://learn.microsoft.com/en-us/azure/ai-services/openai/reference) does
+    /// not support in azure version more than 16 element in a single batch, while openai accepts up to
+    /// 2k. This settings is needed to avoid sending too large batches to the service and having an error.
+    ///
+    /// Openai Has much larger windows as explained here https://platform.openai.com/docs/api-reference/embeddings/create.
+    /// </summary>
+    public int EmbeddingBatchMaxSize { get; set; } = 16;
```

Comment:
A safe default value should be "1". Depending on which model, the region and when the model was deployed, some deployments support only a batch of 1:

> Too many inputs. The max number of inputs is 1.  We hope to increase the number of inputs per request soon. Please contact us through an Azure support request at: https://go.microsoft.com/fwlink/?linkid=2213926 for further questions.

By the way, with one deployment I was able to send batches of 2000+ strings, this is going to be very hard to manage...

Reply:
I know, older deployment with the first model does not support batching, newest embedding have no limit, nevertheless the user should be able to set this value accordingly to actual deployment, I does not see a different solution